samples/Gallery/Pages/AutoCompleteBoxPage.fs

```diff
@@ -337,6 +365,49 @@ module AutoCompleteBoxPage =
                 }
         }
 
+    let searchUsFederalStatesAsync (term: string) (cancellation: CancellationToken) : Task<seq<obj>> =
+        (*  TODO If you debug this while trying completely new search terms in quick succession,
+            you'll notice that the dropdown sometimes gets populated with results for your previous searches.
+
+            It looks to me like some searches don't get cancelled (maybe due to re-rendering caused by the SearchTextChanged event?)
```

Comment:
Yeah this seems to be a syncronization  issue. I guess applying some debouncing will help. We have a `Cmd.debounce` but not sure who to fit in your current sample.

Reply:
I do not have any suggestion on how to improve this ATM

Reply:
I have added a change that wraps the remote search into a type that tracks and can cancel its running search (via a custom `CancellationTokenSource`) to avoid this issue and it seems to work alright. I'll let you know if I have an idea about how to do that on the fly to make the async populator safe to use.